manifests/join.pp

```diff
@@ -23,8 +27,19 @@
     notify {'For Active Directory join to work you must specify the ad_join_ou parameter.':}
   }
   else {
+    if $ad_join_os != undef {
+      $ad_join_os_command = " --os-name=\'${ad_join_os}\'"
+    }
+    if $ad_join_os_version != undef {
+      $ad_join_os_version_command = " --os-version=\'${ad_join_os_version}\'"
+    }
+    if $ad_join_os_service_pack != undef {
+      $ad_join_os_service_pack_command = " --os-service-pack=\'${ad_join_os_service_pack}\'"
+    }
```

Comment:
I like the way you did this and would be nice if #4 would get reworked this way so that the `undef` default for `--domain-controller` doesn't mess with the original behavior of this module (unrelated I know but kind of a reminder to myself that it needs fixed before it gets merged...)